grpc/controller.go

```diff
@@ -0,0 +1,93 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onflow/flow-go/ledger"
+	"google.golang.org/grpc"
+
+	"github.com/awfm9/flow-dps/model"
+)
+
+type Controller struct {
+	state model.State
+}
+
+// NewController creates a controller using the given state store.
+func NewController(state model.State) (*Controller, error) {
+	c := &Controller{
+		state: state,
+	}
+	return c, nil
+}
+
+// GetRegister gets a single register from a given key. Block height can also be given as an optional parameter.
+// If no height is given, the last height present within the state is used.
+func (c *Controller) GetRegister(_ context.Context, req *GetRegisterRequest, _ ...grpc.CallOption) (*Register, error) {
+	state := c.state.Raw()
+
+	height, _ := c.state.Last()
+	if req.Height != nil {
+		height = *req.Height
+	}
+
+	state = state.WithHeight(height)
+
+	value, err := state.Get(req.Key)
+	if err != nil {
+		return nil, fmt.Errorf("could not get register in GRPC API: %w", err)
+	}
+
+	return &Register{
+		Height: height,
+		Key: req.Key,
+		Value: value,
+	}, nil
```

Comment:
I don't know what your take on this is, but I dislike direct initializer returns like this, _especially_ when they are multi-line. I always avoid it. It makes the multiple return values less than obvious. In this case, I would initialize `res` and then return `res, nil`.